### ISSUE TITLE
Active changes feeds not notified when deferred sequences get cached

### ISSUE THREAD

**Issue by adamcfraser** [couchbase/sync_gateway#2626]
There are several places in change_cache.go where we call processEntry for non-document updates (skipped sequence handling, unused sequence handling, recent sequence handling), and we're not doing listener notification based on the results of that call.  The non-document update doesn't require notification, but it's possible that the processEntry call unblocked some pending sequences that DO require notification.

Need to ensure notification happens when a non-empty list of channels is returned from processEntry, in all situations.

**Comment by tleyden** [comment]
Possibly related to https://github.com/couchbase/sync_gateway/issues/2626

**Comment by tleyden** [comment]
Possibly related to https://github.com/couchbase/sync_gateway/issues/2358
